### PR TITLE
Support serialize and deserialize nested custom types

### PR DESCRIPTION
**Pull request by hank121314**
## Summary

This PR fixes: #101.

`Defaults` now support serializing and deserializing nested custom types.

## Thanks

Thanks for your code review 😄 !

### CODE REVIEW

File: Tests/DefaultsTests/DefaultsCustomBridgeTests.swift [hunk 1]
```diff
@@ -148,6 +192,35 @@ final class DefaultsCustomBridge: XCTestCase {
...
+		let key = Defaults.Key<PlainHourMinuteTimeRange>("independentCustomBridgeRecursiveKey", default: range)
+		XCTAssertEqual(Defaults[key].start.hour, range.start.hour)
+		XCTAssertEqual(Defaults[key].start.minute, range.start.minute)
+		XCTAssertEqual(Defaults[key].end.hour, range.end.hour)
+		XCTAssertEqual(Defaults[key].end.minute, range.end.minute)
+		guard let rawValue = UserDefaults.standard.array(forKey: key.name) as? [String] else {
+			XCTFail("rawValue should not be nil")
+			return
+		}
+		XCTAssertEqual(rawValue, ["{\"minute\":0,\"hour\":1}", "{\"minute\":0,\"hour\":2}"])
```
Comment (sindresorhus) [inline Tests/DefaultsTests/DefaultsCustomBridgeTests.swift:208]: You can use `#"x"#` string to avoid the escaping.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Thanks for fixing this. Do you plan more changes after this PR? If not, I will make a release.

**Comment by hank121314** [comment]
> Do you plan more changes after this PR?

I would like to move on to #108.
But I think this feature might be useful, maybe it is worth to make a minor release?

**Comment by sindresorhus** [comment]
> maybe it is worth to make a minor release?

Ah, then it's no rush. We can wait until after that. Another reason to wait a little bit more is that the release would require Xcode 14.1.

**Comment by hank121314** [comment]
> Ah, then it's no rush. We can wait until after that. Another reason to wait a little bit more is that the release would require Xcode 14.1.

Got it! That makes sense. Thanks for the explanation!

**Pull request merged**